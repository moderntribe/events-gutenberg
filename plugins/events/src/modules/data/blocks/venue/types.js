/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_STORE } from '@moderntribe/events/data/utils';

export const SET_VENUE = `${ PREFIX_EVENTS_STORE }/SET_VENUE`;

export const TOGGLE_VENUE_MAP = `${ PREFIX_EVENTS_STORE }/TOGGLE_VENUE_MAP`;
export const SET_VENUE_MAP = `${ PREFIX_EVENTS_STORE }/SET_VENUE_MAP`;

export const TOGGLE_VENUE_MAP_LINK = `${ PREFIX_EVENTS_STORE }/TOGGLE_VENUE_MAP_LINK`;
export const SET_VENUE_MAP_LINK = `${ PREFIX_EVENTS_STORE }/SET_VENUE_MAP_LINK`;
