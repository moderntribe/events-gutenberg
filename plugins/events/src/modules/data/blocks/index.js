/**
 * Internal dependencies
 */
import reducer from './reducers';

export default reducer;
