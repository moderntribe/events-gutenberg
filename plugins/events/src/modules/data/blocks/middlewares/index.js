export * from './dates';
export { default as multiDay } from './multiday';
export { default as allDay } from './allday';
export * from './time';
