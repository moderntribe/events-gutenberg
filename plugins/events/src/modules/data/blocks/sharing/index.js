/**
 * Internal dependencies
 */
import reducer from './reducers';

import * as selectors from './selectors';
import * as actions from './actions';
import * as types from './types';

export default reducer;
export { selectors, actions, types };
