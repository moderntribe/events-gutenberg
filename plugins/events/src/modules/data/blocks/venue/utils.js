/**
 * External dependencies
 */
import { isEmpty, get } from 'lodash';
import { getStateName } from '@moderntribe/events/editor/utils/geo-data';

export const getAddress = ( details = {} ) => {
	const { meta = {} } = details;

	if ( isEmpty( meta ) ) {
		return {};
	}

	return {
		street: get( meta, '_VenueAddress', '' ),
		city: get( meta, '_VenueCity', '' ),
		province: get( meta, '_VenueProvince', '' ),
		zip: get( meta, '_VenueZip', '' ),
		country: get( meta, '_VenueCountry', '' ),
	};
};

export const getCoordinates = ( details = {} ) => {
	const { meta = {} } = details;
	const { _VenueLat = '', _VenueLng = '' } = meta;
	const lat = parseFloat( _VenueLat );
	const lng = parseFloat( _VenueLng );

	return {
		lat: isNaN( lat ) ? null : lat,
		lng: isNaN( lng ) ? null : lng,
	};
};

const setDefault = ( value, defaultValue ) => value === '' ? defaultValue : value;

/**
 * Get Venue Country
 */
export function getVenueCountry( meta ) {
	let country = get( meta, '_VenueCountry', '' );

	if ( '' === country ) {
		const defaultCountry = tribe_blocks_editor_defaults.venueCountry;
		country = defaultCountry !== undefined || array.length > 0 ? defaultCountry[1] : '';
	}
	return country;
}

/**
 * Get Venue State/Province
 */
export function getVenueStateProvince( meta ) {
	let stateProvince = get( meta, '_VenueStateProvince', '' );

	if ( '' === stateProvince ) {

		const country = getVenueCountry( meta );

		if (
			'US' === country
			|| 'United States' === country
		) {
			stateProvince = getStateName( 'US', tribe_blocks_editor_defaults.venueState );
		} else {
			stateProvince = tribe_blocks_editor_defaults.venueProvince;
		}
	}
	return stateProvince;
}
