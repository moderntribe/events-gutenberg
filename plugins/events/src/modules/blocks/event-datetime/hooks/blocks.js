/**
 * External dependencies
 */
import React from 'react';
import PropTypes from 'prop-types';

/**
 * Internal Dependencies
 */
import { constants } from '@moderntribe/common/data/plugins';
import { PluginBlockHooks } from '@moderntribe/common/components';

const PLUGIN_TEMPLATES = {
	[ constants.EVENTS_PRO_PLUGIN ]: [
		[ 'tribe/event-pro-recurring', {}],
	],
};

const PluginDateTimeBlockHooks = () => {
	return (
		<PluginBlockHooks
			pluginTemplates={ PLUGIN_TEMPLATES }
			templateLock="all"
		/>
	);
};

PluginDateTimeBlockHooks.propTypes = {};

export default PluginDateTimeBlockHooks;
