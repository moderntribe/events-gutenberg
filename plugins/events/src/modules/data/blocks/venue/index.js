/**
 * Internal dependencies
 */
import reducer from './reducers';
import * as selectors from './selectors';
import * as actions from './actions';
import * as types from './types';
import * as utils from './utils';

export default reducer;
export { selectors, actions, types, utils };
