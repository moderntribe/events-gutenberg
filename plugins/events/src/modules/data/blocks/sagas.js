
/**
 * External dependencies
 */
import { store } from '@moderntribe/common/store';

/**
 * Internal dependencies
 */
import * as price from '@moderntribe/events/data/blocks/price';
import * as venue from '@moderntribe/events/data/blocks/venue';

[
	price.sagas,
	venue.sagas,
].forEach( sagas => store.run( sagas ) );
