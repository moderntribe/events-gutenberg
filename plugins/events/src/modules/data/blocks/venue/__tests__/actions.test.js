/**
 * External dependencies
 */
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
/**
 * Internal dependencies
 */
import { actions } from '@@plugins/events/data/blocks/venue';

const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );

describe( '[STORE] - Venue actions', () => {
	test( 'action to set the venue', () => {
		expect( actions.setVenue( 99 ) ).toMatchSnapshot();
	} );

	test( 'action to set the showMap', () => {
		expect( actions.setShowMap( false ) ).toMatchSnapshot();
		expect( actions.setShowMap( true ) ).toMatchSnapshot();
	} );

	test( 'action to set the showMapLink', () => {
		expect( actions.setShowMapLink( true ) ).toMatchSnapshot();
		expect( actions.setShowMapLink( false ) ).toMatchSnapshot();
	} );

	test( 'action to set the venue removal', () => {
		expect( actions.removeVenue() ).toMatchSnapshot();
	} );

	test( 'action to toggle the venue map', () => {
		expect( actions.toggleVenueMap() ).toMatchSnapshot();
	} );

	test( 'action to toggle the venueMapLink', () => {
		expect( actions.toggleVenueMapLink() ).toMatchSnapshot();
	} );
} );

describe( '[STORE] - Venue thunk actions', () => {
	test( 'Initial set of state', () => {
		const store = mockStore( {} );

		const get = jest.fn( ( key ) => {
			const values = {
				venue: 99,
				showMap: true,
				showMapLink: true,
			};
			return values[ key ];
		} );

		store.dispatch( actions.setInitialState( { get } ) );

		expect( get ).toHaveBeenCalled();
		expect( get ).toHaveBeenCalledTimes( 3 );
		expect( store.getActions() ).toMatchSnapshot();
	} );
} );

