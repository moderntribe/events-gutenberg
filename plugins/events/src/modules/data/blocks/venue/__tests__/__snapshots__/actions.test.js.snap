// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Venue actions action to set the showMap 1`] = `
Object {
  "payload": Object {
    "showMap": false,
  },
  "type": "SET_VENUE_MAP",
}
`;

exports[`[STORE] - Venue actions action to set the showMap 2`] = `
Object {
  "payload": Object {
    "showMap": true,
  },
  "type": "SET_VENUE_MAP",
}
`;

exports[`[STORE] - Venue actions action to set the showMapLink 1`] = `
Object {
  "payload": Object {
    "showMapLink": true,
  },
  "type": "SET_VENUE_MAP_LINK",
}
`;

exports[`[STORE] - Venue actions action to set the showMapLink 2`] = `
Object {
  "payload": Object {
    "showMapLink": false,
  },
  "type": "SET_VENUE_MAP_LINK",
}
`;

exports[`[STORE] - Venue actions action to set the venue 1`] = `
Object {
  "payload": Object {
    "venue": 99,
  },
  "type": "SET_VENUE",
}
`;

exports[`[STORE] - Venue actions action to set the venue removal 1`] = `
Object {
  "payload": Object {
    "venue": undefined,
  },
  "type": "SET_VENUE",
}
`;

exports[`[STORE] - Venue actions action to toggle the venue map 1`] = `
Object {
  "type": "TOGGLE_VENUE_MAP",
}
`;

exports[`[STORE] - Venue actions action to toggle the venueMapLink 1`] = `
Object {
  "type": "TOGGLE_VENUE_MAP_LINK",
}
`;

exports[`[STORE] - Venue thunk actions Initial set of state 1`] = `
Array [
  Object {
    "payload": Object {
      "venue": 99,
    },
    "type": "SET_VENUE",
  },
  Object {
    "payload": Object {
      "showMap": true,
    },
    "type": "SET_VENUE_MAP",
  },
  Object {
    "payload": Object {
      "showMapLink": true,
    },
    "type": "SET_VENUE_MAP_LINK",
  },
]
`;
